### PR TITLE
[WIP]Registery for callbacks

### PR DESCRIPTION
**Pull request by rustworthy**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [ ] update .d.ts typings: not applicable

Changes:
- 'calls' may sound just a little bit confusing, so I suggest renaming this storage to 'callbacksRegistry';
- suggesting here to subclass Registry from Map and add a utility method  `getOrSetDefault`, which makes the code tighter in loader;

### CODE REVIEW

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
Use class syntax please

File: lib/node.js [hunk 1]
```diff
@@ -14,7 +23,7 @@ const prepareImports = (byteCode) => {
     if (!module) module = expected[entry.module] = {};
     module[entry.name] = (...args) => {
       const name = entry.name.slice(0, -CALLBACK_LEN);
-      const callbacks = calls.get(name);
+      const callbacks = callbacksRegistry.get(name);
       if (!callbacks) return;
       const callback = callbacks.shift();
       if (!callback) return;
```
Comment (tshemsedinov) [inline lib/node.js]: We may hide mentioned lines behind `callbacksRegistry.getNextCallback(name);`

File: lib/node.js [hunk 2]
```diff
@@ -38,11 +47,7 @@ const load = async (fileName, importObject = {}) => {
     }
     exports[name] = (...args) => {
       if (typeof args.at(-1) !== 'function') return fn(...args);
-      let callbacks = calls.get(name);
-      if (!callbacks) {
-        callbacks = [];
-        calls.set(name, callbacks);
-      }
+      const callbacks = callbacksRegistry.getOrSetDefault(name, []);
```
Comment (tshemsedinov) [inline lib/node.js]: - Do not pass empty array to this method
- Rename `.getOrSetDefault(name)` to `.getCallbacks(name)`

**Comment by rustworthy** [comment]
Closed in favour of [this one](https://github.com/tshemsedinov/wasm-import/pull/26)

**Pull request closed without merging**